File: workshops/InversionOfControl/tobi/toasterlauncher/src/main/java/com/novoda/toasterlauncher/AppComponent.java
package com.novoda.toasterlauncher;


import com.novoda.iockata.core.di.BuildersModule;

import dagger.Component;
import dagger.android.AndroidInjector;
import dagger.android.support.AndroidSupportInjectionModule;

@Component(modules = { AndroidSupportInjectionModule.class, AppModule.class, BuildersModule.class })
public interface AppComponent extends AndroidInjector<LauncherApplication> {

    @Component.Builder
    abstract class Builder extends AndroidInjector.Builder<LauncherApplication> {
    }
}

File: workshops/InversionOfControl/default/app/src/main/java/com/novoda/iockata/di/AppComponent.kt
package com.novoda.iockata.di

import dagger.Component
import dagger.android.AndroidInjector
import dagger.android.support.AndroidSupportInjectionModule

@Component(modules = arrayOf(
        AndroidSupportInjectionModule::class,
        AppModule::class,
        BuildersModule::class))
abstract class AppComponent: AndroidInjector<IoCLauncherApplication> {

    @Component.Builder
    abstract class Builder : AndroidInjector.Builder<IoCLauncherApplication>()

}
